module_utils/network_lsr/ethtool.py

```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+import array
+import struct
+import fcntl
+import socket
+
+from .utils import Util
+
+ETHTOOL_GPERMADDR = 0x00000020
+SIOCETHTOOL = 0x8946
+
+
+def get_perm_addr(ifname):
+    """
+    Return the Permanent address value for the specified interface using the
+    ETHTOOL_GPERMADDR ioctl command.
+
+    Please for further documentation, see:
+    https://github.com/torvalds/linux/blob/master/include/uapi/linux/ethtool.h#L734
+    https://github.com/torvalds/linux/blob/master/include/uapi/linux/ethtool.h#L1388
+    https://git.kernel.org/pub/scm/network/ethtool/ethtool.git/tree/ethtool.c#n4172
+    """
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sockfd = sock.fileno()
+        ifname = ifname.encode("utf-8")
+        if len(ifname) > 16:
```

Comment:
Please define a constant for this `IFNAMSZ` and use it here and for the struct format strings. Please also use one for the perm address size and check that this will also work for infiniband devices that have a longer address. Thanks.

Reply:
Done, thanks! It will work with infiniband devices.